src/components/CookieProvider.ts

```diff
@@ -0,0 +1,19 @@
+import { Ajax } from 'src/libs/ajax';
+import { AbortOption } from 'src/libs/ajax/data-provider-common';
+import { azureCookieReadyStore, cookieReadyStore } from 'src/libs/state';
+
+export interface CookieProvider {
+  invalidateCookies: (options?: AbortOption) => Promise<void>;
+}
+
+export const cookieProvider: CookieProvider = {
```

Comment:
I really like this general pattern, but the name is a bit too generic.... instead of cookieProvider, maybe leoAuthSyncProvider?

Reply:
also, I think this should live somewhere other then src/components.... if it's Leo-specific, maybe somewhere in src/analysis?